src/tests/ftest/deployment/ior_per_rank.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2018-2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+
+from avocado.core.exceptions import TestFail
+
+from ior_test_base import IorTestBase
+from ior_utils import IorCommand, IorMetrics
+from general_utils import percent_change
+
+
+class IorPerRank(IorTestBase):
+    # pylint: disable=attribute-defined-outside-init
+    # pylint: disable=too-few-public-methods
+    # pylint: disable=too-many-ancestors
+    """Test Class Description: Test class to run ior per rank basis and
+                               collect read/write data to identify bad
+                               nodes.
+    :avocado: recursive
+    """
+
+    def execute_ior_thread(self, rank):
+        """
+        Method to execute ior for different transfer sizes,
+        collect the performance numbers and match with the
+        expected values.
+        Args:
+            rank(str): Daos server rank on which ior needs to be run.
+        """
+
+        self.log.info("Running Test on rank: %s", rank)
+        # create the pool on specified rank.
+        self.create_pool([rank])
+        # execute ior on given rank for different transfer sizes and collect the results
+        for idx, _ in enumerate(self.transfer_sizes):
+            self.ior_cmd.transfer_size.update(self.transfer_sizes[idx])
+            dfs_out = self.run_ior_with_pool(fail_on_warning=self.log.info)
+            dfs_perf = IorCommand.get_ior_metrics(dfs_out)
+
+            # Destroy container, to be sure we use newly created container in next iteration
+            self.container.destroy()
+            self.container = None
+
+            # gather actual and expected perf data to be compared
+            if idx == 0:
+                dfs_max_write = float(dfs_perf[0][IorMetrics.Max_MiB])
+                dfs_max_read = float(dfs_perf[1][IorMetrics.Max_MiB])
+                actual_write_x = percent_change(dfs_max_write, self.expected_bw)
+                actual_read_x = percent_change(dfs_max_read, self.expected_bw)
+            else:
+                dfs_max_write = float(dfs_perf[0][IorMetrics.Max_OPs])
+                dfs_max_read = float(dfs_perf[1][IorMetrics.Max_OPs])
+                actual_write_x = percent_change(dfs_max_write, self.expected_iops)
+                actual_read_x = percent_change(dfs_max_read, self.expected_iops)
```

Comment:
To be clear, the intent is to check BW for the first iteration only, and then IOPs for every other iteration? Is this because of the 256B transfer size?

Reply:
That's correct. We are looking for IOPS for 256B ts.